kube-runtime/src/scheduler.rs

```diff
@@ -16,11 +17,18 @@ use tokio_util::time::delay_queue::{self, DelayQueue};
 pub struct ScheduleRequest<T> {
     pub message: T,
     pub run_at: Instant,
```

Comment:
Could we impose a new restriction on `T` that allows grabbing the reason, rather than caching it in here? I.e. `impl ToString for ReconcileRequest` and rely on it here?